pkg/ipam/poolallocator/allocator.go

```diff
@@ -252,6 +250,35 @@ func (a *IPPoolAllocator) removeIPAddressState(ipPool *v1alpha2.IPPool, ip net.I
 
 }
 
+// getExistingAllocation looks up the existing IP allocation for a Pod network interface, and
+// returns the IP address and SubnetInfo if found.
+func (a *IPPoolAllocator) getExistingAllocation(podOwner *v1alpha2.PodOwner) (net.IP, *v1alpha2.SubnetInfo, error) {
+	ip, err := a.HasContainer(podOwner.ContainerID, podOwner.IFName)
+	if err != nil {
+		return nil, nil, err
+	}
+	if ip == nil {
+		return nil, nil, nil
+	}
+
+	ipPool, allocators, err := a.getPoolAndInitIPAllocators()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	index := -1
+	for i, allocator := range allocators {
+		if allocator.Has(ip) {
```

Comment:
This returns true if IP is in given range, regardless if it is allocated. Perhaps we should rename it to avoid confusion.

Reply:
What is the right name in your mind? But anyway it is not relevant to this PR, and I prefer not to change it in this PR.

Reply:
Yes, its just a thought not relevant for the PR. Perhaps `belongs` would be a more fitting name.